### ISSUE TITLE
Facebook ERROR: unable to download video data: HTTP Error 404: Not Found (On a specific video)

### ISSUE THREAD

**Issue by BigDru** [ytdl-org/youtube-dl#19161]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.01.30.1*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2019.01.30.1**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
youtube-dl -vic -w -o C:/Users/BigDru/Videos/%(title)s.%(ext)s https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-vic', '-w', '-o', 'C:/Users/BigDru/Videos/%(title)s.%(ext)s', 'https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2019.01.30.1
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.17134
[debug] exe versions: none
[debug] Proxy map: {}
[facebook] 1942829622685298: Downloading webpage
[debug] Default format spec: best/bestvideo+bestaudio
[debug] Invoking downloader on 'https://scontent.fyto1-1.fna.fbcdn.net/v/t66.18014-6/10000000_360583261355566_5781567373328691887_n.mp4?_nc_cat=109&efg=eyJ2ZW5jb2RlX3RhZyI6Im9lcF9oZCJ9&_nc_ht=scontent.fyto1-1.fna&oh=645b387995dd865eab0c73af6091c77e&oe=5CFA4698'
ERROR: unable to download video data: HTTP Error 404: Not Found
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\YoutubeDL.py", line 1912, in process_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\YoutubeDL.py", line 1851, in dl
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\downloader\common.py", line 364, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\downloader\http.py", line 341, in real_download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\downloader\http.py", line 109, in establish_connection
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmp36zpl0n4\build\youtube_dl\YoutubeDL.py", line 2215, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 580, in http_response
  File "C:\Python\Python34\lib\urllib\request.py", line 508, in error
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Python\Python34\lib\urllib\request.py", line 588, in http_error_default
urllib.error.HTTPError: HTTP Error 404: Not Found
...
<end of log>
```

---

---

### Description of your *issue*, suggested solution and other information

Resolution of this specific facebook video URL turns up a 404 error. Another video I tried worked fine.

**Comment by alexvong1995** [comment]
@BigDru
I cannot reproduce the error. The download completes successfully for me:
```
alexvong1995@debian:~$ torsocks youtube-dl -vic -w https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-vic', '-w', 'https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.01.27
1549570933 WARNING torsocks[16159]: [syscall] Unsupported syscall number 217. Denying the call (in tsocks_syscall() at syscall.c:568)
[debug] Python version 3.7.0 (CPython) - Linux-4.19.0-2-amd64-x86_64-with-debian-buster-sid
[debug] exe versions: ffmpeg 4.1, ffprobe 4.1
[debug] Proxy map: {}
[facebook] 1942829622685298: Downloading webpage
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://video-lht6-1.xx.fbcdn.net/v/t42.1790-2/10000000_2309276759340813_6106046037107933184_n.mp4?_nc_cat=110&efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfdjRfaHE1X2ZyYWdfMl92aWRlbyJ9&_nc_ht=video-lht6-1.xx&oh=6d1d3c6b0c94cfca8b1bb232f9b9f982&oe=5C5CD286'
[download] Destination: Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2309276749340814v.mp4
[download] 100% of 14.87MiB in 01:04
[debug] Invoking downloader on 'https://video-lht6-1.xx.fbcdn.net/v/t42.1790-2/47301521_2166217156959795_2071376050125799424_n.mp4?_nc_cat=111&efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfYXVkaW9fYWFjcF82NF9mcmFnXzJfYXVkaW8ifQ==&_nc_ht=video-lht6-1.xx&oh=70972d9741a5d722e1fe6ee5493f660c&oe=5C5CBA57'
[download] Destination: Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2166217153626462ad.m4a
[download] 100% of 1.82MiB in 00:10
[ffmpeg] Merging formats into "Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.mp4"
[debug] ffmpeg command line: ffmpeg -y -i 'file:Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2309276749340814v.mp4' -i 'file:Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2166217153626462ad.m4a' -c copy -map 0:v:0 -map 1:a:0 'file:Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.temp.mp4'
Deleting original file Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2309276749340814v.mp4 (pass -k to keep)
Deleting original file Played Für Elise for my daughter Emilie the other day, she told me she liked ...-1942829622685298.f2166217153626462ad.m4a (pass -k to keep)

```
Does the error persist?

**Comment by BigDru** [comment]
Yes. The error still persists. Even after updating to 2019.02.08. 
I also tried it on a different computer and through a VPN linked to the US (originally was downloading from Canada). 

Interesting that for you the url resolves to a different target. 
https://video-lht6-1.xx.fbcdn.net/v/t42.1790-2/
vs
https://scontent.fyto1-1.fna.fbcdn.net/v/t66.18014-6/

I also noticed that you're using Python v 3.7.0. Probably doesn't have any effect though.

**Comment by alexvong1995** [comment]
@BigDru

BigDru <notifications@github.com> writes:

> Yes. The error still persists. Even after updating to 2019.02.08.
> I also tried it on a different computer and through a VPN linked to the US (originally was
> downloading from Canada).
>
Could it be that it is also blocked in the US? For me, I always run
> torsocks youtube-dl ...
instead of
> youtube-dl ...
to bypass geo restriction using tor and torsocks.

However, the internet seems to suggest that torsocks is not available on
windows. In this case, I guess one should first set up a running tor
SOCKS5 proxy (https://www.deepdotweb.com/2014/05/23/use-tor-socks5-proxy/).
Then, follow this issue (https://github.com/rg3/youtube-dl/issues/402)
and run
> youtube-dl --proxy "socks5://localhost:9150" ...
to download.

I don't use windows though, so it may not work at all.

> Interesting that for you the url resolves to a different target.
> https://video-lht6-1.xx.fbcdn.net/v/t42.1790-2/
> vs
> https://scontent.fyto1-1.fna.fbcdn.net/v/t66.18014-6/
>
I didn't notice it! I think it is because
CDN (https://en.wikipedia.org/wiki/Content_delivery_network) will try to
fetch the content from the best location.

> I also noticed that you're using Python v 3.7.0. Probably doesn't have any effect though.
>
I would be surprised if a 404 is caused by using an old python version,
but never say never in the world of backward compatibility ...

**Comment by barsnick** [comment]
Accessing from Germany, I can confirm that the "best" format gives a 404, while most others (too lazy to test them all) seem to work. Here are the available formats:
```
[barsnick@paradise youtube-dl]$ python3 -m youtube_dl -F https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/
[facebook] 1942829622685298: Downloading webpage
[info] Available formats for 1942829622685298:
format code               extension  resolution note
2166217153626462ad        m4a        audio only [eng] DASH audio   65k , m4a_dash container, mp4a.40.5 (48000Hz)
2242693562655099v         mp4        256x144    [eng] DASH video   33k , mp4_dash container, avc1.4D401E, 25fps, video only
2038468652914870v         mp4        426x240    [eng] DASH video   92k , mp4_dash container, avc1.4D401E, 25fps, video only
215528269339063vd         mp4        640x360    [eng] DASH video  168k , mp4_dash container, avc1.4D401E, 25fps, video only
2079245129032963v         mp4        852x480    [eng] DASH video  242k , mp4_dash container, avc1.4D401F, 25fps, video only
739742513048796v          mp4        1280x720   [eng] DASH video  272k , mp4_dash container, avc1.4D401F, 25fps, video only
2309276749340814v         mp4        1920x1080  [eng] DASH video  533k , mp4_dash container, avc1.4D4028, 25fps, video only
dash_sd_src               mp4        unknown
dash_sd_src_no_ratelimit  mp4        unknown
dash_hd_src               mp4        unknown    (best)
```
Here's default (works):
```
[barsnick@paradise youtube-dl]$ python3 -m youtube_dl -v https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.01.02
[debug] Git HEAD: 9dfd27538
[debug] Python version 3.7.2 (CPython) - Linux-4.20.6-200.fc29.x86_64-x86_64-with-fedora-29-Twenty_Nine
[debug] exe versions: ffmpeg 4.0.3, ffprobe 4.0.3
[debug] Proxy map: {}
[facebook] 1942829622685298: Downloading webpage
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://video.fhaj2-1.fna.fbcdn.net/v/t42.1790-2/10000000_2309276759340813_6106046037107933184_n.mp4?_nc_cat=110&efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfdjRfaHE1X2ZyYWdfMl92aWRlbyJ9&_nc_ht=video.fhaj2-1.fna&oh=1342cbdffbbfec231728badecf786f12&oe=5C6B0CB6'
[download] Destination: Beethoven “Für Elise” Paul Barton, FEURICH 218 piano-1942829622685298.f2309276749340814v.mp4
[download]  53.8% of 14.87MiB at  5.26MiB/s ETA 00:01^C
ERROR: Interrupted by user
```
And here's manual selection of "best" a.k.a. dash_hd_src (fails with 404):
```
[barsnick@paradise youtube-dl]$ python3 -m youtube_dl -v -f dash_hd_src https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-f', 'dash_hd_src', 'https://www.facebook.com/PaulBartonPiano/videos/1942829622685298/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.01.02
[debug] Git HEAD: 9dfd27538
[debug] Python version 3.7.2 (CPython) - Linux-4.20.6-200.fc29.x86_64-x86_64-with-fedora-29-Twenty_Nine
[debug] exe versions: ffmpeg 4.0.3, ffprobe 4.0.3
[debug] Proxy map: {}
[facebook] 1942829622685298: Downloading webpage
[debug] Invoking downloader on 'https://scontent.fhaj2-1.fna.fbcdn.net/v/t66.18014-6/10000000_360583261355566_5781567373328691887_n.mp4?_nc_cat=109&efg=eyJ2ZW5jb2RlX3RhZyI6Im9lcF9oZCJ9&_nc_ht=scontent.fhaj2-1.fna&oh=826a5dd1838d2596b943a68f1ae7a8da&oe=5D21D398'
ERROR: unable to download video data: HTTP Error 404: Not Found
Traceback (most recent call last):
  File "/home/barsnick/Development/youtube-dl/youtube_dl/YoutubeDL.py", line 1909, in process_info
    success = dl(filename, info_dict)
  File "/home/barsnick/Development/youtube-dl/youtube_dl/YoutubeDL.py", line 1848, in dl
    return fd.download(name, info)
  File "/home/barsnick/Development/youtube-dl/youtube_dl/downloader/common.py", line 364, in download
    return self.real_download(filename, info_dict)
  File "/home/barsnick/Development/youtube-dl/youtube_dl/downloader/http.py", line 341, in real_download
    establish_connection()
  File "/home/barsnick/Development/youtube-dl/youtube_dl/downloader/http.py", line 109, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "/home/barsnick/Development/youtube-dl/youtube_dl/YoutubeDL.py", line 2212, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib64/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib64/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

(Sorry, I'm on a local branch working on FB changes, can't switch. That's why my target filename is different.)

**Comment by remitamine** [comment]
Server side problem, nothing to be done on youtube-dl side.
